discovery_engine/lib/src/domain/models/active_search.dart

```diff
@@ -12,28 +12,46 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:equatable/equatable.dart';
 import 'package:hive/hive.dart'
     show HiveType, HiveField, TypeAdapter, BinaryReader, BinaryWriter;
+import 'package:xayn_discovery_engine/src/api/models/active_search.dart' as api;
 import 'package:xayn_discovery_engine/src/domain/repository/type_id.dart'
     show searchTypeId, searchByTypeId;
 
-part 'active_search.freezed.dart';
 part 'active_search.g.dart';
 
 /// [ActiveSearch] represents attributes of a performed search.
-@freezed
-class ActiveSearch with _$ActiveSearch {
-  @HiveType(typeId: searchTypeId)
-  const factory ActiveSearch({
-    @HiveField(0) required String searchTerm,
-    @HiveField(1) required int requestedPageNb,
-    @HiveField(2) required int pageSize,
-    @HiveField(3, defaultValue: SearchBy.query) required SearchBy searchBy,
-  }) = _ActiveSearch;
-
-  factory ActiveSearch.fromJson(Map<String, Object?> json) =>
-      _$ActiveSearchFromJson(json);
+@HiveType(typeId: searchTypeId)
+class ActiveSearch with EquatableMixin {
+  @HiveField(0)
+  final String searchTerm;
+  @HiveField(1)
+  final int requestedPageNb;
+  @HiveField(2)
+  final int pageSize;
+  @HiveField(3, defaultValue: SearchBy.query)
+  final SearchBy searchBy;
+
+  const ActiveSearch({
+    required this.searchTerm,
+    required this.requestedPageNb,
+    required this.pageSize,
+    required this.searchBy,
+  });
+
+  ActiveSearch nextPageSearch() => ActiveSearch(
```

Comment:
❓ is it required (by hive or for other reasons) to create a new object? otherwise we could just mutate the current object by incrementing the `requestedPageNb`. this would be more in line with the updating character of the function as it is used in the search manager.

Reply:
Also this data is a basically a POD, it should not contain any logic about what the next page is.

Reply:
> question is it required (by hive or for other reasons) to create a new object? 

No, just less likely to cause problems due to accidental shared references in a GCed language.
And at least new/before the fields had been final.

> Also this data is a basically a POD, it should not contain any logic about what the next page is.

This is roughly the same design as `Document` which also has small helpers (in form of computed fields/getters).

Before we used `foo.copyWith(requestedPageNb: foo.requestedPageNb + 1)` ad-hoc, but that's a bit prone to mistakes.

Furthermore given that this type already contains the concept of paging it's not like it's some logic specific to some implementation (or at least not more then the fields already are).

If you have a better place to put the method I will move it there. I mean I did consider an extension trait but it's just not worth the additional overhead IMHO.